content/v2.0/application/booting.md

```diff
@@ -0,0 +1,164 @@
+
+<p class="notice">
+  A prepared application is no less able to do its job than a booted one, it will just lazily load only the components it needs to perform a particular task on demand.
+</p>
+
+## Hanami.boot
+
+When you call `Hanami.boot` (or use `require "hanami/boot"`) Hanami will do everything it does when `Hanami.prepare` is called, but then go further, starting each of your app's providers, while also eagerly loading all of your app's components up front.
+
+This is useful when you want to incur all initialization costs at boot time, such as when preparing an app to serve web requests.
+
+Booting is the approach taken in Hanami's standard Puma setup. Thus, in Hanami's `config.ru` file you will see:
+
+```ruby
+# frozen_string_literal: true
+
+require "hanami/boot"
+
+run Hanami.app
+```
+
+## Stepping through the boot process
+
+Purely as an exercise, we can explore both preparing and booting by starting `irb` in a directory containing a Hanami project. This is not something that's needed in day to day development with Hanami - it's useful here as a demonstration of the booting behaviour.
+
+Assuming we have run `hanami new bookshelf` to generate a new app (see [Getting started](/v2.0/introduction/getting-started/) for a full guide to creating your first Hanami application), let's create a hello world component in `app/hello_world.rb`:
+
+```ruby
+# app/hello_world.rb
+
+# frozen_string_literal: true
+
+module Bookshelf
+  class HelloWorld
+    def say_hello
+      "Hello world!"
+    end
+  end
+end
+```
+
+And now, in our project directory, let's run `irb`:
+
+```ruby
+bundle exec irb
+
+irb(main)>
+```
+
+Typing `Hanami` here will result in a `NameError`:
+
+```ruby
+irb(main)> Hanami
+(irb):1:in `<main>': uninitialized constant Hanami (NameError)
+```
+
+Let's address that by running `require "hanami"`.
+
+Once the Hanami module is available, calling `Hanami.setup` will load the Hanami app defined in `config/app.rb`:
+
+```ruby
+irb(main)> require "hanami"
+=> true
+
+irb(main)> Hanami.setup
+=> Bookshelf::App
+
+irb(main)> Hanami.app
+=> Bookshelf::App
+```
+
+We can now ask our app whether it's prepared or booted:
+
+```ruby
+irb(main)> Hanami.app.prepared?
+=> false
+
+irb(main)> Hanami.app.booted?
+=> false
+```
+
+We can also see what components are registered with the app by calling `#keys`:
+
+```ruby
+irb(main)> Hanami.app.keys
+=> []
+```
+
+This makes sense, as the app hasn't even been prepared yet, meaning component registration hasn't begun!
+
+Let's prepare the app now, then ask what component keys are registered:
+
+```ruby
+irb(main)> Hanami.prepare
+=> Bookshelf::App
+
+irb(main)> Hanami.app.prepared?
+=> true
+
+irb(main)> Hanami.app.keys
+=> ["settings", "notifications"]
+```
+
+Two components are now present: `settings` and `notifications`.
+
+`notifications` is an instance of `Dry::Monitor::Notifications` that's registered early in the prepare process to support app-wide notifications.
```

Comment:
It might be worth adding a couple of words here to qualify "notifications"... maybe something like:

> to support inter-component notifications (for now mostly a framework-internal concern). 

I'm a little concerned about this object spooking people a bit, not the least of which because it adopts a name that might quite likely be wanted by app authors for their own components.

Reply:
Thanks, adjusted!